src/klein/resource.py

```diff
@@ -280,3 +280,39 @@ def write_response(r):
         d.addCallback(write_response).addErrback(log.err, _why="Unhandled Error writing response")
 
         return server.NOT_DONE_YET
+
+
+
+class KleinHTTPRequest(server.Request):
+
+    def getArg(self, key):
+        """
+        Get a single arg value.
+
+        @raises KeyError: If key doesn't exist
+        @raises ValueError: If there is more than 1 value
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        value = self.args[key]
+        if len(value) != 1:
+            raise ValueError('Too many values for: {0}'.format(key))
+        return value[0]
+
+    def getArgs(self, key):
+        """
+        Get the list of values for a key.
+
+        @return: L{list} of L{bytes}
+        """
+        key = ensure_utf8_bytes(key)
+        return self.args.get(key, [])
+
+
+
+class KleinSite(server.Site):
```

Comment:
Is this intended to be public?  Do we need to use it for all Klein applications now?  That seems like a pretty big change to quietly be foisting on all Klein applications. (That implies `.app.resource()` won't work in a normal server any more, or will randomly be missing some methods.)

If it is intended to be public, some docstrings seems like a bare minimum.  Also, `Site` is an old-style class, so this should definitely have an `, object` at the end of its inheritance hierarchy.